google/data_source_google_project_services.go

```diff
@@ -0,0 +1,35 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleProjectServices() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceGoogleProjectServices().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "project")
+
+	return &schema.Resource{
+		Read:   dataSourceGoogleProjectServicesRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceGoogleProjectServicesRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	if v, ok := d.GetOk("project"); ok {
```

Comment:
This is exactly what getProject does, so you can get rid of the first part of this if statement and leave everything that's in the `else` block.